### PR TITLE
Require specifying what package symlinks you want created

### PR DESCRIPTION
**Pull request by nburka**

### CODE REVIEW

File: lib/symlinks.js [hunk 1]
```diff
@@ -1,19 +1,27 @@
...
  * package working directories instead of installed composer packages
  */
 function setup(err, progress, complete) {
   if (fs.existsSync(paths.vendors)) {
-    fs.readdirSync(paths.vendors).forEach((packageName) => {
+    packages.forEach((packageName) => {
       const packageLinkPath = path.join(paths.vendors, packageName);
-      if (fs.lstatSync(packageLinkPath).isDirectory()) {
+
+      if (fs.existsSync(packageLinkPath) &&
```
Comment (nburka) [inline lib/symlinks.js]: should we throw an error if the package doesn't exist? or output a message to the command line?

File: lib/symlinks.js [hunk 2]
```diff
@@ -1,19 +1,27 @@
 const gutil = require('gulp-util');
+const minimist = require('minimist');
 const fs = require('fs');
 const path = require('path');
 const paths = require('./paths');
 
 const suffix = '.original';
+const options = minimist(process.argv.slice(2), {
```
Comment (nburka) [inline lib/symlinks.js]: I'm basically duplicating this from the gulpfile. Is there a better way to pass an argument? This seems ok, but maybe there's a better way?

File: lib/symlinks.js (hunk 1 above)
Comment (gauthierm) [reply lib/symlinks.js]: a warning message seems appropriate

File: lib/symlinks.js (hunk 2 above)
Comment (gauthierm) [reply lib/symlinks.js]: parse the options in gulpfile.js. Pass the arguments to symlinks.task.setup and symlinks.task.teardown.

Comment (gauthierm) [reply lib/symlinks.js]: gulp tasks do not accept arguments (see the setup task) so you'll need to create a helper task. From the symlink file, export functions that take arguments. In the gulpfile.js, wrap them in a function and then use that in dependencies:

```js
function symlinkSetupTask() {
  return symlinks.setup(options.symlinks);
}
function symlinkTeardownTask() {
  return symlinks.teardown(options.symlinks);
}
```

Then when defining tasks:
```js
gulp.task('setup-symlinks', symlinkSetupTask);
...
```

File: lib/symlinks.js (hunk 1 above)
Comment (nburka) [reply lib/symlinks.js]: I now show a message

File: lib/symlinks.js [hunk 3]
```diff
@@ -9,11 +9,15 @@ const suffix = '.original';
  * Replaces vendor package directories with symlinks that point to active
  * package working directories instead of installed composer packages
  */
-function setup(err, progress, complete) {
+function setup(symlinks, err, progress, complete) {
```
Comment (gauthierm) [inline lib/symlinks.js]: by convention, the error should be the first argument in nodejs functions

File: lib/symlinks.js [hunk 4]
```diff
@@ -37,6 +41,10 @@ function setup(err, progress, complete) {
             );
           }
         }
+      } else {
+        gutil.log(
+          gutil.colors.red('Package ' + packageLinkPath + ' not found')
```
Comment (gauthierm) [inline lib/symlinks.js]: use template strings here:
```js
`Package ${packageLinkPath} not found`
```

**Pull request merged**